TradeSkillMaster_Crafting/Modules/SpellNames2IDs.lua

```diff
@@ -1,3583 +1,32 @@
-	["Skullfish Soup"] = 43707,
-	["Stormchops"] = 43758,
-	["Broiled Bloodfin"] = 43761,
-	["Spicy Hot Talbuk"] = 43765,
-	["Kibler's Bits"] = 43772,
-	["Delicious Chocolate Cake"] = 43779,
-	["Hot Apple Cider"] = 45022,
-	["Mammoth Meal"] = 45549,
-	["Shoveltusk Steak"] = 45550,
-	["Worm Delight"] = 45551,
-	["Roasted Worg"] = 45552,
-	["Rhino Dogs"] = 45553,
-	["Great Feast"] = 45554,
-	["Mega Mammoth Meal"] = 45555,
-	["Tender Shoveltusk Steak"] = 45556,
-	["Spiced Worm Burger"] = 45557,
-	["Very Burnt Worg"] = 45558,
-	["Mighty Rhino Dogs"] = 45559,
-	["Smoked Rockfin"] = 45560,
-	["Grilled Bonescale"] = 45561,
-	["Sauteed Goby"] = 45562,
-	["Grilled Sculpin"] = 45563,
-	["Smoked Salmon"] = 45564,
-	["Poached Nettlefish"] = 45565,
-	["Pickled Fangtooth"] = 45566,
-	["Poached Northern Sculpin"] = 45567,
-	["Firecracker Salmon"] = 45568,
-	["Baked Manta Ray"] = 45569,
-	["Imperial Manta Steak"] = 45570,
-	["Spicy Blue Nettlefish"] = 45571,
-	["Captain Rumsey's Lager"] = 45695,
-	["Charred Bear Kabobs"] = 46684,
-	["Juicy Bear Burger"] = 46688,
-	["Kungaloosh"] = 53056,
-	["Northern Stew"] = 57421,
-	["Fish Feast"] = 57423,
-	["Spicy Fried Herring"] = 57433,
-	["Rhinolicious Wormsteak"] = 57434,
-	["Critter Bites"] = 57435,
-	["Hearty Rhino"] = 57436,
-	["Snapper Extreme"] = 57437,
-	["Blackened Worg Steak"] = 57438,
-	["Cuttlesteak"] = 57439,
-	["Spiced Mammoth Treats"] = 57440,
-	["Blackened Dragonfin"] = 57441,
-	["Dragonfin Filet"] = 57442,
-	["Tracker Snacks"] = 57443,
-	["Dalaran Clam Chowder"] = 58065,
-	["Tasty Cupcake"] = 58512,
-	["Last Week's Mammoth"] = 58521,
-	["Bad Clams"] = 58523,
-	["Haunted Herring"] = 58525,
-	["Gigantic Feast"] = 58527,
-	["Small Feast"] = 58528,
-	["Worg Tartare"] = 62350,
-	["Clamlette Magnifique"] = 64054,
-	["Black Jelly"] = 64358,
-	["Bread of the Dead"] = 65454,
-	["Candied Sweet Potato"] = 66034,
-	["Cranberry Chutney"] = 66035,
-	["Pumpkin Pie"] = 66036,
-	["Slow-Roasted Turkey"] = 66037,
-	["Spice Bread Stuffing"] = 66038,
-
-
-
--- First Aid
-	["First Aid"] = 3273,
-	["Linen Bandage"] = 3275,
-	["Heavy Linen Bandage"] = 3276,
-	["Wool Bandage"] = 3277,
-	["Heavy Wool Bandage"] = 3278,
-	["Anti-Venom"] = 7394,
-	["Silk Bandage"] = 7928,
-	["Heavy Silk Bandage"] = 7929,
-	["Strong Anti-Venom"] = 7935,
-	["Mageweave Bandage"] = 10840,
-	["Heavy Mageweave Bandage"] = 10841,
-	["Runecloth Bandage"] = 18629,
-	["Heavy Runecloth Bandage"] = 18630,
-	["Powerful Anti-Venom"] = 23787,
-	["Netherweave Bandage"] = 27032,
-	["Heavy Netherweave Bandage"] = 27033,
-	["Frostweave Bandage"] = 45545,
-	["Heavy Frostweave Bandage"] = 45546
-}
+local idCache = setmetatable({}, {
+    -- Use weak references both for key and values
+    __mode = 'kv',
+    -- This is called only if the key has not been found in the table
+    __index = function(self, key)
+        -- ID to check is stored in the table itself, so it can get GC'ed
+        local id = rawget(self, '__id') or 0
+        while id < 100000 do
```

Comment:
Ascension has tradeskill spell ids higher than this. 

Not sure of a good solution, but it can be checked for with 

`if AscensionUI then`